pkg/perf/cache.go

```diff
@@ -0,0 +1,49 @@
+package perf
+
+import (
+	"context"
+	"encoding/json"
+	"time"
+
+	"github.com/pkg/errors"
+)
+
+// TestResultData the result test schema
+type TestResultData struct {
+	TestName   string
+	TestNumber uint64
+	Result     interface{}
+}
+
+// GetRequest the get request struct
+type GetRequest struct {
+	TestName   string `json:"test_name"`
+	TestNumber uint64 `json:"test_number"`
+}
+
+// CacheResult set result in redis
+func (pm *PerformanceMonitor) CacheResult(ctx context.Context, resultKey string, resultData TestResultData) error {
```

Comment:
Why is this a `public` method. I don't expect anyone should ever call this method by himself. this should an internal method. I see many attributes and methods that are `public` for no obvious reason. It make the code subtle to abuse and bad usage. And allows bad design